clients/src/test/java/org/apache/kafka/common/network/SelectorTest.java

```diff
@@ -127,8 +127,10 @@ public void testConnectionRefused() throws Exception {
         ServerSocket nonListeningSocket = new ServerSocket(0);
         int nonListeningPort = nonListeningSocket.getLocalPort();
         selector.connect(node, new InetSocketAddress("localhost", nonListeningPort), BUFFER_SIZE, BUFFER_SIZE);
-        while (selector.disconnected().contains(node))
+        while (selector.disconnected().containsKey(node)) {
+            assertEquals(ChannelState.NOT_CONNECTED, selector.disconnected().get(node));
```

Comment:
In the other PR, the an issue was raised that `disconnected` only gets populated if there are in flight requests. And if the broker is misconfigured, we never have in-flight request. How have we fixed that?